java/WordCountOptimized.java

```diff
@@ -217,33 +207,66 @@ public static int tokenizeAndSubmitBlock(TObjectIntCustomHashMap<byte[]> counts,
         }
         return 0;
     }
-    
-    private static void submitWord(TObjectIntCustomHashMap<byte[]> m, byte[] word) {
-        m.adjustOrPutValue(word, 1, 1);        
+
+    private static final int TABLE_SIZE = 1024 * 1024;
+    private static final int TABLE_MASK = TABLE_SIZE - 1;
+
+    static class WordMap {
+        final byte[][] keys = new byte[TABLE_SIZE][];
+        final int[] values = new int[TABLE_SIZE];
+        int size = 0;
     }
-    
+
+    private static int submitWord(byte[][] keys, int[] values, int size,
```

Comment:
@svanoort funnily I didn't read that, just implemented exactly what you said. However it was on the surface. now it is interesting how much faster is this.